js/client-teambuilder.js

```diff
@@ -2059,7 +2063,7 @@
 
 			if (this.curTeam.gen > 1) {
 				buf += '<div class="formrow"><label class="formlabel">Gender:</label><div>';
-				if (template.gender) {
+				if (template.gender && this.curTeam.format !== 'balancedhackmons') {
 					var genderTable = {'M': "Male", 'F': "Female", 'N': "Genderless"};
 					buf += genderTable[template.gender];
 				} else {
```

Comment:
`N` means "genderless" and not "random" in Hackmons, that needs to be fixed too.